### PR TITLE
Migrate AppCompatPreferenceActivity to kotlin

### PR DESCRIPTION
**Pull request by lukstbit**
## Purpose / Description

Migrate AppCompatPreferenceActivity.java to kotlin following the kotlin migration guide.

## How Has This Been Tested?

Ran the usual tests.

## Checklist

- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/ui/AppCompatPreferenceActivity.kt [hunk 1]
```diff
@@ -14,125 +14,113 @@
...
-import androidx.annotation.Nullable;
-import androidx.appcompat.app.ActionBar;
-import androidx.appcompat.app.AppCompatDelegate;
-import androidx.appcompat.widget.Toolbar;
-import android.view.MenuInflater;
-import android.view.View;
-import android.view.ViewGroup;
-
-import com.ichi2.anki.AnkiDroidApp;
+@file:Suppress("DEPRECATION")
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/ui/AppCompatPreferenceActivity.kt:17]: Could you move the comment here as well (and remove it from below):

`// TODO Tracked in https://github.com/ankidroid/Anki-Android/issues/5019`

**Review by david-allison — COMMENTED** [review]
Looks great! One tiny change

File: AnkiDroid/src/main/java/com/ichi2/ui/AppCompatPreferenceActivity.kt (hunk 1 above)
Comment (lukstbit) [reply AnkiDroid/src/main/java/com/ichi2/ui/AppCompatPreferenceActivity.kt]: Done!

**Review by david-allison — APPROVED** [review]
LGTM

**Review by mikehardy — APPROVED** [review]
LGTM thank you!

**Pull request merged**